Sources_v3/StreamChatUI/ChatChannelList/ChatChannelListVC.swift

```diff
@@ -13,6 +13,23 @@ open class ChatChannelListVC<ExtraData: UIExtraDataTypes>: UIViewController,
     public var controller: _ChatChannelListController<ExtraData>!
     public var uiConfig: UIConfig<ExtraData> = .default
     public var didSelectChannel: (_ChatChannel<ExtraData>) -> Void = { _ in }
+    public var didSelectCurrentUser: () -> Void = {}
+    public var didSelectCreateNewChannel: () -> Void = {}
+
+    private lazy var userAvatarView: CurrentChatUserAvatarView<ExtraData> = {
+        let avatar = uiConfig.channelList.currentUserAvatarView.init()
+        avatar.translatesAutoresizingMaskIntoConstraints = false
+        avatar.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(didTapOnCurrentUserAvatar)))
```

Comment:
This is not ideal because we lose the pressed state. Maybe we should rather wrap this whole thing in a button?

Reply:
hm 🤔 You suggest to keep `CurrentUserAvatar: AvatarView` and wrap it into the button OR have `CurrentUserAvatar: UIButton`?

Reply:
Wrapping it

Reply:
resolved by having a `UIControl` subclass